File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        HashMap<String, Integer> map = new HashMap<>();
        for (String name : participant) {
            if (map.containsKey(name)) {
                map.replace(name, map.get(name) + 1);
            }
            else {
                map.put(name, 1);
            }
        }
        for (String name : completion) {
            if (map.containsKey(name)) {
                map.replace(name, map.get(name) - 1);
            }
        }
        String answer = "";
        for (String name : map.keySet()) {
            if (map.get(name) >= 1) {
                answer = name;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string, int> m;
    for (string name : participant) m[name]++;
    for (string name : completion) m[name]--;
    for (auto pair : m) {
        if (pair.second) answer = pair.first;
    }
    return answer;
}